### PR TITLE
weact stm32h743vit6 usbconsole fix

### PR DESCRIPTION
**Pull request by TOMASHEVSKIYIVAN**
fix previous problemc

added defconfig for weact-stm32h743:nshusb

## Summary
added full support of usb console 
## Impact
new debug experience

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * **Why is the change necessary?** What problem does it fix? 
    * **What functional part of the code is being changed?** Be specific (e.g., USB driver, networking stack, build system).
    * **How does the change work?** Provide technical details about the implementation.
    * **Related Issues/PRs:**  Link any relevant issues or pull requests. 

* **Impact:**
   * Most of the impact sections are left blank.  You **must** address each point, even if the answer is "NO" (provide a brief explanation if necessary).

* **Testing:**
    * **Insufficient Detail:** "fix previous problemc" and "added defconfig" are not descriptive enough.
    * **Missing Logs:** You need to provide actual testing logs from before and after the change to demonstrate the issue and the fix.

**Recommendations:**

1. **Expand the Summary:**  Clearly articulate the problem, solution, and technical details of your change.
2. **Complete the Impact Section:**  Thoroughly address each impact point, even if the answer is "NO."
3. **Provide Detailed Testing Information:** Include:
    * Specific build host and target details.
    * Clear and complete testing logs from before and after the change.
    * Steps to reproduce the issue (if applicable).

**Remember:** A well-written PR makes it easier for maintainers to understand and review your changes, increasing the likelihood of your contribution being accepted.

File: boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h [hunk 1]
```diff
@@ -46,6 +49,54 @@
 #  endif
 #endif
 
+/* Can't support USB host or device features if USB OTG FS is not enabled */
+
+#ifndef CONFIG_STM32H7_OTGFS
+#  undef HAVE_USBDEV
```
Comment (xiaoxiang781216) [inline boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:55]: why need define HAVE_XXX maros?

Comment (TOMASHEVSKIYIVAN) [reply boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:55]: if not configured OTG FS then there should be no chance to use it in compile

Comment (xiaoxiang781216) [reply boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:55]: but I think no code reference HAVE_XXXX in nuttx code base.

Comment (TOMASHEVSKIYIVAN) [reply boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:55]: I found code-like in nuttx/boards/arm/stm32h7/linum-stm32h753bi

Comment (xiaoxiang781216) [reply boards/arm/stm32h7/weact-stm32h743/src/weact-stm32h743.h:55]: OK

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request closed without merging**